docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+
+
+![image alt text](images/image_2.jpg "DAG for Pure Style SQL Application")
+
+<a name="fusion-style-sql-application"></a>
+## Example 2: Fusion Style SQL Application 
+
+As described in Pure Style SQL application, we can use different data sources as source and sink while developing Apex Applications with Calcite. This example will describe how we can develop Apex application with Apex stream as abstract table for SQL query. Actual source code can be found [here](https://github.com/apache/apex-malhar/blob/master/demos/sql/src/main/java/org/apache/apex/malhar/sql/sample/FusionStyleSQLApplication.java).
+
+
+```java      
+// Define Kafka Input Operator for reading data from Kafka
+KafkaSinglePortInputOperator kafkaInput = dag.addOperator("KafkaInput", 
+                                                           KafkaSinglePortInputOperator.class);
+
+kafkaInput.setInitialOffset("EARLIEST");
+
+// Add CSVParser
+CsvParser csvParser = dag.addOperator("CSVParser", CsvParser.class);
+dag.addStream("KafkaToCSV", kafkaInput.outputPort, csvParser.in);
+```
+
+Once we define DAG with KafkaInputOperator and CSVParser, it can parse data from Kafka topic. Upto this point, this is a regular Apex application without SQL. After this, we can register the output of CSVParser as a table using [StreamEndpoint](https://github.com/apache/apex-malhar/blob/master/sql/src/main/java/org/apache/apex/malhar/sql/table/StreamEndpoint.java) to run a SQL statement. This way we can develop applications in fusion style where the DAG is part SQL and part regular apex DAG.
+
+The following code will describe how we can define StreamEndpoint. 
+ 
+```java  
+SQLExecEnvironment sqlEnv = sqlEnv.getEnvironment();
+Map<String, Class> fieldMapping = ImmutableMap.<String, Class>of("RowTime", Date.class,
+                                                                 "id", Integer.class,
+                                                                 "Product", String.class,
+                                                                 "units", Integer.class);
+sqlEnv = sqlEnv.registerTable("FROMCSV", new StreamEndpoint(csvParser.out, fieldMapping));
+```
+
+To read existing data stream, we need to register it as a table with SQL execution environment with the name of the table and StreamEndpoint. StreamEndpoint can serve as input table or output table in SQL. For input table configuration we need to initialise StreamEndpoint with immediate upstream operator's output port and fieldMapping or POJO class for input tuple. For output table configuration, we need to initialise StreamEndpoint with immediate downstream operator's input port and fieldMapping or POJO class for output tuple. Once we register StreamEndpoint as a table with a name in SQL Execution Environment, it can be used as a table in SQL statement similar to other endpoints.
```

Comment:
output port and fieldMapping or POJO class for input tuple => output port and fieldMapping or POJO class for input tuple (as shown above).